Functions/BIDS/xASL_bids_GenerateParticipantsJSON.m

```diff
@@ -0,0 +1,152 @@
+            jsonParticipants.csf_vol.Units = 'liter';
+        case {'gm_icv_ratio', 'gm_icvratio'}
+            jsonParticipants.gm_icv_ratio.Description = 'Ratio of gray matter volume to the intracranial volume of the participant at this visit, proxy of gray matter atrophy';
+            jsonParticipants.gm_icv_ratio.Units = 'ratio';
+        case {'gmwm_icv_ratio', 'gmwm_icvratio'}
+            jsonParticipants.gmwm_icv_ratio.Description = 'Ratio of parenchymal volume (gray matter + WM volume) to the intracranial volume of the participant at this visit, proxy of wholebrain atrophy';
+            jsonParticipants.gmwm_icv_ratio.Units = 'ratio';
+        case 'wmh_vol'
+            jsonParticipants.wmh_vol.Description = 'White matter hyperintensity volume of the participant at this visit';
+            jsonParticipants.wmh_vol.Units = 'liter';
+        case 'wmh_count'
+            jsonParticipants.wmh_count.Description = 'Number of non-connected white matter hyperintensities of the participant at this visit';
+            jsonParticipants.wmh_count.Units = 'integer';
+        case 'meanmotion'
+            jsonParticipants.motion.Description = 'Mean net displacement vector difference (head motion) of this ASL scan';
+            jsonParticipants.wmh_count.Units = 'mm RMS';
+        otherwise
+
+
+            %% 4. Parse ASL-related keys
+            if ~isempty(regexpi(listKeys{iList}, '(cbf|att|m0|control|tex|itt)'))
+                descriptionBIDS = lower(listKeys{iList});
+
+
+                %% 4a. Replace key abbreviations with verbose description
+                descriptionBIDS = strrep(descriptionBIDS, 'cbf', 'cerebral blood flow');
+                descriptionBIDS = strrep(descriptionBIDS, 'att', 'arterial transit time');
+                descriptionBIDS = strrep(descriptionBIDS, 'm0', 'M0 reference intensity');
+                descriptionBIDS = strrep(descriptionBIDS, 'meancontrol', 'control');
+                descriptionBIDS = strrep(descriptionBIDS, 'itt', 'intravoxel transit time');
+                descriptionBIDS = strrep(descriptionBIDS, 'tex', 'time of exchange');
+                
+                descriptionBIDS = strrep(descriptionBIDS, '_b', ' bilateral');
+                descriptionBIDS = strrep(descriptionBIDS, '_l', ' left');
+                descriptionBIDS = strrep(descriptionBIDS, '_r', ' right');
+                descriptionBIDS = strrep(descriptionBIDS, '_aca', ' anterior cerebral artery territory');
+                descriptionBIDS = strrep(descriptionBIDS, '_mca', ' middle cerebral artery territory');
+                descriptionBIDS = strrep(descriptionBIDS, '_pca', ' posterior cerebral artery territory');
+                descriptionBIDS = strrep(descriptionBIDS, '_gm', ' gray matter');
+                descriptionBIDS = strrep(descriptionBIDS, '_wm', ' white matter');
+                descriptionBIDS = strrep(descriptionBIDS, 'deepwm white matter', 'deep white matter');
```

Comment:
is this line correct? replacing "deepwm white matter" with "deep white matter"?

Reply:
Yes, because it takes the ROI and the TissueMasking, so e.g., Total & GM, or deepWM & WM in this case, which we replace with deep WM for niceness.

TotalGM was renamed to Total, because it is simply the whole brain which we mask with GM.
In the case of deep WM, it is a standard space deep WM ROI which we mask with the WM.

Reply:
No then it would be 

'_deepwm white matter' -> ' deep WM'

Anyway this seems to work :)